apps/coscrad-frontend/src/components/tree-of-knowledge/category-tree-node.tsx

```diff
@@ -0,0 +1,103 @@
+import {
+    AggregateType,
+    CategorizableType,
+    ICategoryTreeViewModel,
+    ICompositeIdentifier,
+} from '@coscrad/api-interfaces';
+import { isNullOrUndefined } from '@coscrad/validation-constraints';
+import {
+    ChevronRight as ChevronRightIcon,
+    ExpandMore as ExpandMoreIcon,
+} from '@mui/icons-material';
+import {
+    Accordion,
+    AccordionDetails,
+    AccordionSummary,
+    Collapse,
+    List,
+    ListItemButton,
+    ListItemIcon,
+    ListItemText,
+} from '@mui/material';
+import { useContext, useState } from 'react';
+import { buildDataAttributeForAggregateDetailComponent } from '../../utils/generic-components/presenters/detail-views/build-data-attribute-for-aggregate-detail-component';
+import { CategorizablesOfMultipleTypeContainer } from '../higher-order-components';
+import { thumbnailCategorizableDetailPresenterFactory } from '../resources/factories/thumbnail-categorizable-detail-presenter-factory';
+import { CategoryTreeUXContext } from './category-tree-ui-context';
+
+interface TreeNodeProps {
+    id: string;
+    label: string;
+    members: ICompositeIdentifier<CategorizableType>[];
+    children: ICategoryTreeViewModel<CategorizableType>[];
+    wrapTree: ({
```

Comment:
Was this always passed in as a callback? This is a bit funky, but might have been done for good reasons.

Reply:
No, this is new